### PR TITLE
Add view-logs anchor to publishing progress indicator

### LINKED ISSUE #1796: Add hyperlink into deployment progress (in home view) to view publisher log

### PR DESCRIPTION
**Pull request by sagerb**
## Intent

Resolves #1796 

Add an anchor to show the Publishing Logs while deployment is in progress.

## Type of Change

- - [ ] Bug Fix 
- - [x] New Feature 
- - [ ] Breaking Change 
- - [ ] Documentation 
- - [ ] Refactor 
- - [ ] Tooling 

## Approach

Add an anchor only shown with the progress indicator, and execute the existing command to show the publishing log.

2024-06-27 at 3 06 PM

## Directions for Reviewers

Deploy and click on anchor while deployment in progress.

## Checklist

- [ ] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.

### CODE REVIEW

**Review by mmarchetti — COMMENTED** [review]
Code LGTM, and I verified locally.

I'm not sure about the appearance. Could it be a small button? Or, if it's a link, I feel like it should have more spacing from the text above it. 

I like the `View Log` wording. Since we have the same command in the ellipsis menu, can we use the same text to describe it?

File: extensions/vscode/webviews/homeView/src/components/EvenEasierDeploy.vue [hunk 1]
```diff
@@ -105,7 +105,14 @@
         <div class="deployment-in-progress-container">
           <div class="progress-container">
             <vscode-progress-ring class="progress-ring" />
-            Deployment in Progress...
+            <div class="progress-desc">
+              <div>Deployment in Progress...</div>
+              <div>
+                <a href="" role="button" @click="onViewPublishingLog"
+                  >View Log</a
```
Comment (dotNomad) [inline extensions/vscode/webviews/homeView/src/components/EvenEasierDeploy.vue:112]: I'll also mention we have a mix of "Log" and "Logs". We should probably unify under singular or plural.

**Comment by dotNomad** [comment]
> I'm not sure about the appearance. Could it be a small button? Or, if it's a link, I feel like it should have more spacing from the text above it.

I think a button would be too large. The buttons we have are how VSCode styles buttons, unless you are talking about an icon button, but I think the text is more helpful here.

We could definitely add a bit of margin between the in progress text and the logs link though.

File: extensions/vscode/webviews/homeView/src/components/EvenEasierDeploy.vue (hunk 1 above)
Comment (sagerb) [reply extensions/vscode/webviews/homeView/src/components/EvenEasierDeploy.vue:112]: I searched for "logs" and if they were referencing a target string seen by the user, I changed them to log. See commit: https://github.com/posit-dev/publisher/pull/1896/commits/5b5da72c1296e5518b6fc8c8ed522d129d68b6a2

Comment (dotNomad) [reply extensions/vscode/webviews/homeView/src/components/EvenEasierDeploy.vue:112]: Fantastic. Thanks for unifying that.

**Comment by sagerb** [comment]
> Code LGTM, and I verified locally.
> 
> I'm not sure about the appearance. Could it be a small button? Or, if it's a link, I feel like it should have more spacing from the text above it.
> 
> I like the `View Log` wording. Since we have the same command in the ellipsis menu, can we use the same text to describe it?

We have multiple logs to target within the ellipsis menus. I think it would be confusing if we had the publisher log listed as `View Log`, but the other one

> Code LGTM, and I verified locally.
> 
> I'm not sure about the appearance. Could it be a small button? Or, if it's a link, I feel like it should have more spacing from the text above it.
> 
> I like the `View Log` wording. Since we have the same command in the ellipsis menu, can we use the same text to describe it?

I went with the link because I also thought a button would be too big there. I'll add a bit of margin.

On your wording request, we have multiple logs available to be seen within the ellipsis menu. Are you asking for me to switch to the phrase "View" rather than "Go to"? For context, here is the ellipsis menu currently:
2024-07-01 at 12 05 PM

**Comment by dotNomad** [comment]
> On your wording request, we have multiple logs available to be seen within the ellipsis menu.

I read it as making the wording consistent so instead of "View Log" it would be "View Publishing Log" for the link and the context menu. I agree that consistency would be nice.

I'll let @mmarchetti confirm if I was reading that correctly.

**Comment by mmarchetti** [comment]
> I think a button would be too large

Yes, a typical button would be. I was thinking along the lines of the buttons in the Extensions panel:

CleanShot 2024-07-01 at 15 21 05

I am also fine with keeping it as a link. As to wording, I would like to be consistent in the names we use for the same action. Seems like the options we've used or considered for showing a view are:
* Show ___
* Go to ___
* View ___
* (and I guess VSCode's default `Focus on ___` in the command palette)

Of the 3, I think my preference for VSCode-local views would be `Show` because it shows an existing view. The View Content and View Content Log actions open new browser tabs and could retain different wording to differentiate them. I'm totally open to other options though.

**Comment by sagerb** [comment]
> > I'm not sure about the appearance. Could it be a small button? Or, if it's a link, I feel like it should have more spacing from the text above it.
> 
> I think a button would be too large. The buttons we have are how VSCode styles buttons, unless you are talking about an icon button, but I think the text is more helpful here.
> 
> We could definitely add a bit of margin between the in progress text and the logs link though.

Here's a 5px top margin added:
2024-07-01 at 12 41 PM

**Comment by sagerb** [comment]
> Of the 3, I think my preference for VSCode-local views would be `Show` because it shows an existing view. The View Content and View Content Log actions open new browser tabs and could retain different wording to differentiate them. I'm totally open to other options though.

OK, I've updated to the following:
- Use of "Show" verb is for views which are present within VSCode
- Use of "View" verb is for an external view, with a qualifier for knowing it is external (typically "on Connect")

See https://github.com/posit-dev/publisher/pull/1896/commits/d50fee625b33494d786170c1eb0e592f0610c834

**Comment by sagerb** [comment]
> Yes, a typical button would be. I was thinking along the lines of the buttons in the Extensions panel:

@mmarchetti , are you thinking you want to replace the ellipsis menu then, with that dropdown menu?

**Comment by mmarchetti** [comment]
I was just showing a small button that could be used in place of the `Show Log` link. But, I'm fine with this as is. It looks better with the added margin.

**Review by mmarchetti — APPROVED** [review]

**Review by dotNomad — APPROVED** [review]
Additional changes look great!

**Pull request merged**